llvm-general/src/LLVM/General/Internal/Attribute.hs

```diff
@@ -120,6 +143,9 @@ instance DecodeM DecodeAST A.FA.FunctionAttribute FFI.FunctionAttribute where
        else do
          enum <- liftIO $ FFI.functionAttributeKindAsEnum a
          case enum of
+           [functionAttributeKindP|AllocSize|] -> do
+             (x,y) <- liftIO $ FFI.attributeGetAllocSizeArgs a
```

Comment:
As noted at the definition of attributeGetAllocSizeArgs, it really belongs in this directory. Please try to keep the encode/decode logic parallel.